finetune/lora.py

```diff
@@ -110,21 +110,20 @@ def main(fabric: L.Fabric, data_dir: Path, checkpoint_dir: Path, out_dir: Path):
     )
     checkpoint_path = checkpoint_dir / "lit_model.pth"
     fabric.print(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}")
-    with fabric.init_module(empty_init=False):
+    with fabric.init_module(empty_init=True):
         model = GPT(config)
-        model.apply(model._init_weights)  # for the LoRA weights
-    with lazy_load(checkpoint_path) as checkpoint:
-        # strict=False because missing keys due to LoRA weights not contained in state dict
-        model.load_state_dict(checkpoint, strict=False)
-
     mark_only_lora_as_trainable(model)
 
     fabric.print(f"Number of trainable parameters: {num_parameters(model, requires_grad=True):,}")
     fabric.print(f"Number of non trainable parameters: {num_parameters(model, requires_grad=False):,}")
-    trainable_params = [p for p in model.parameters() if p.requires_grad]
 
+    model = fabric.setup_module(model)
+    trainable_params = [p for p in model.parameters() if p.requires_grad]
     optimizer = torch.optim.AdamW(trainable_params, lr=learning_rate, weight_decay=weight_decay)
-    model, optimizer = fabric.setup(model, optimizer)
+    optimizer = fabric.setup_optimizers(optimizer)
+    
+    # strict=False because missing keys due to LoRA weights not contained in state dict
+    fabric.load_raw(checkpoint_path, model, strict=False)
```

Comment:
Using `load_raw` is only equivalent with regular FSDP.

This PR would break TPUs until `XLAFSDP` supports `load_raw` and `lazy_load`.

If using other strategies, it would also mean `lazy_load` is no longer enabled because it was only added to regular FSDP with fabric.